src/main/java/net/dv8tion/jda/core/managers/impl/AudioManagerImpl.java

```diff
@@ -103,9 +105,15 @@ public void openAudioConnection(VoiceChannel channel)
                 return;
 
             final int userLimit = channel.getUserLimit(); // userLimit is 0 if no limit is set!
-            if (!self.hasPermission(channel, Permission.MANAGE_CHANNEL) && userLimit > 0 && userLimit <= channel.getMembers().size())
-                throw new PermissionException(Permission.MANAGE_CHANNEL,
-                        "Unable to connect to VoiceChannel due to userlimit! Requires permission MANAGE_CHANNEL to bypass");
+            final long perms = PermissionUtil.getImplicitPermission(channel, self);
```

Comment:
Calculate the implicit perms inside of the below if statement to avoid unnecessary permission calculations.